lib/yelp/responses/base.rb

```diff
@@ -0,0 +1,19 @@
+module Yelp
+  module Response
+    class Base
+      def initialize(json)
+        json.each do |key, value|
+          instance_variable_set("@#{key}", value)
+        end
+      end
+
+      private
+
+      def parse(json, klass)
+        return json.collect { |j| klass.new(j) } if json.class == Array
+        return klass.new(json) if json
```

Comment:
Maybe consider `response_class` or the pythonic `cls` instead of `klass`, non blocking.